apps/src/code-studio/pd/foorm/library_editor/FoormLibrarySaveBar.jsx

```diff
@@ -125,47 +95,77 @@ class FoormSaveBar extends Component {
     this.state = {
       confirmationDialogBeingShownName: null,
       isSaving: false,
-      showNewFormSave: false,
-      formName: null,
-      formCategory: null
+      showNewLibraryQuestionSave: false,
+      libraryQuestionName: null,
+      libraryName: null,
+      formsAppearedIn: [],
+      libraryCategory: null
     };
   }
 
-  updateQuestionsUrl = () =>
-    `/foorm/forms/${this.props.formId}/update_questions`;
-
-  publishUrl = () => `/foorm/forms/${this.props.formId}/publish`;
+  updateQuestionUrl = () =>
+    `/foorm/library_questions/${this.props.libraryQuestionId}`;
 
   handleSave = () => {
     this.setState({isSaving: true});
     this.props.setSaveError(null);
-    if (this.props.isFormPublished) {
-      // show a warning if in published mode
-      this.setState({
-        confirmationDialogBeingShownName: confirmationDialogNames.save
-      });
-    } else if (this.props.formId === null || this.props.formId === undefined) {
-      // if this is not an existing form, show new form save modal
-      this.setState({showNewFormSave: true});
-    } else {
-      this.save(this.updateQuestionsUrl());
+
+    if (
+      this.props.libraryQuestionId === null ||
+      this.props.libraryQuestionId === undefined
```

Comment:
Could replace `this.props.libraryQuestionId === null || this.props.libraryQuestionId === undefined` with `!this.props.libraryQuestionId` as long as libraryQuestionId isn't ever 0